src/components/molecules/UserProfilePicture/UserReactions.scss

```diff
@@ -0,0 +1,115 @@
+@import "scss/constants";
+
+// TODO: should we move these to using CSS --variables rather than these SCSS vars
+$reactionSize: max(2.5vh, 2vw);
+$reactionOffset: calc(-1 * #{$reactionSize} / 2);
+$shoutBounceTranslateXOffset: 50%;
+
+.UserReactions {
+  // Note: we need this to ensure the reactions use this container for their positioning
+  position: relative;
+
+  width: fit-content;
+  height: fit-content;
+
+  &--reaction-left {
+    .UserReactions__reaction {
+      left: $reactionOffset;
+    }
+
+    .UserReactions__shout {
+      right: 50%;
+      transform-origin: right center;
+      animation: shout-bounce--left 5s ease;
+    }
+  }
+
+  &--reaction-right {
+    .UserReactions__reaction {
+      right: $reactionOffset;
+    }
+
+    .UserReactions__shout {
+      left: 50%;
+      transform-origin: left center;
+      animation: shout-bounce--right 5s ease;
+    }
+  }
+
+  &__reaction {
+    position: absolute;
+    top: $reactionOffset;
+
+    z-index: z(user-reactions-emoji);
+
+    // Note: emoji reactions are sized with font-size
+    font-size: $reactionSize;
+
+    animation: reaction-pulse 4s ease-in-out infinite;
+  }
+
+  &__shout {
+    position: absolute;
+    bottom: 0;
+
+    z-index: z(user-reactions-shout);
+
+    width: max-content;
+    max-width: 20em;
+    padding: 6px 10px;
+
+    background-color: rgba(255, 255, 255, 1);
+    color: $black;
+    font-size: 20px;
+    border-radius: 10px;
```

Comment:
I think that we should come up with a convention on how we approach border-radiuses/font-sizes/paddings/margins units. px/rem/em/??? + maybe we can create a styleguide with predefined values of those? Like the golden ratio for margins/padding. @mcflyok wdyt?

Reply:
Definitely agree RE: standardising this + common helpers in `scss/constants.scss` or similar rather than defining it arbitrarily all over the codebase. I believe we already have some variables for this (though it's not always clear when they should be applied, so often I feel like we end up sticking with the legacy definitions by default)

@sunny-viktoryia Made some changes in https://github.com/sparkletown/sparkle/pull/1328 to introduce `$border-radius--small`, etc

- https://github.com/sparkletown/sparkle/blob/bec9ea233a7130a836079592bff2df7aae3a3734/src/scss/constants.scss#L49-L50

Also added SCSS mixins for things like scrollbar styling, since it was repeated all over the codebase:

- https://github.com/sparkletown/sparkle/blob/bec9ea233a7130a836079592bff2df7aae3a3734/src/scss/constants.scss#L128-L148

I'd probably like us to use SCSS variables/mixins/functions/etc where it makes sense to. CSS `--variables` where it makes sense to as well.

My main 'desire' is to keep styling changes to styling; i'm generally not really a fan of 'utility CSS classes' that end up 'infecting' the structural layer of the app code/JS, rather than including it in a specific CSS class 'tied' to that feature. That said, I am open to hearing/being convinced otherwise. I think my main 'concern' here is that if we are applying a specific class to a thing, that specific class should have all of the relevant styles. It shouldn't have to also be combined with additional utility classes in the JSX. That's where mixins/etc become useful IMO. Though in cases where an element doesn't 'need' a specific class for it, and the only thing we want to do is something 'generic', then utility classes might make sense there.

Reply:
(captured the above in https://github.com/sparkletown/internal-sparkle-issues/issues/4#issuecomment-824454619)

Reply:
Going to call this OOS for now in the interest of landing the main fixes in the PR (otherwise will spend too much time on micro polish)

We can follow up this PR in future to use standard values across the codebase + update all the locations at once.